crypto/rsa/rsa_ameth.c

```diff
@@ -118,7 +118,18 @@ static int rsa_pub_decode(EVP_PKEY *pkey, X509_PUBKEY *pubkey)
 
 static int rsa_pub_cmp(const EVP_PKEY *a, const EVP_PKEY *b)
 {
-    if (BN_cmp(b->pkey.rsa->n, a->pkey.rsa->n) != 0
+    /*
+     * Don't check the public/private key, this is mostly for smart
+     * cards.
+     */
+    if (((a->type == EVP_PKEY_RSA)
```

Comment:
This check is probably unnecessary. By virtue of being in this function we must have an RSA key type.